v2/compiler/dag.go

```diff
@@ -180,35 +180,50 @@ func (c *workflowCompiler) addDAGDriverTemplate() string {
 
 func addImplicitDependencies(dagSpec *pipelinespec.DagSpec) error {
 	for _, task := range dagSpec.GetTasks() {
+		wrap := func(err error) error {
+			return fmt.Errorf("failed to add implicit deps: %w", err)
+		}
+		addDep := func(producer string) error {
+			_, ok := dagSpec.GetTasks()[producer]
+			if !ok {
```

Comment:
Can we combine them as
`_, ok := dagSpec.GetTasks()[producer];if !ok {
// handle not ok
}`

Reply:
sure, will do